include/orcania.h

```diff
@@ -10,113 +10,141 @@
 #ifndef __ORCANIA_H__
 #define __ORCANIA_H__
 
-#ifndef U_DISABLE_JANSSON
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef O_EXTERN
+# ifdef O_SHARED_LIBRARY
+#  ifdef _WIN32
+#   define O_EXTERN __declspec(dllexport)
+#  else
+#   define O_EXTERN extern
+#  endif
+# else
+#  ifdef O_STATIC_LIBRARY
+#   define O_EXTERN
+#  else
+#   define O_EXTERN extern
+#  endif
+# endif
+#endif
+
+#ifndef ORCANIA_VERSION
+#define ORCANIA_VERSION LIBRARY_VERSION
+#endif
+
+#ifdef USE_JANSSON
 #include <jansson.h>
 #endif
 
-#define ORCANIA_VERSION 1.1.1
+/**
+ * Return the library version.
+ */
+O_EXTERN const char *o_version();
```

Comment:
What is it for?

Reply:
It is very, very required to export the API functions to other compilers (Pascal, Python, C# etc.). I don't know how to reproduce a demo showing the relevance, but you can see it present in many popular libraries ([mhd](https://github.com/Karlson2k/libmicrohttpd/blob/master/src/include/microhttpd.h#L2119), [curl](https://github.com/curl/curl/blob/master/include/curl/easy.h#L28), [uv](https://github.com/libuv/libuv/blob/v1.x/include/uv.h#L369) etc.). When I remove it, for example, I can't call Orcania's functions from Delphi compiler. Regarding `o_version()`, it is useful for who wants to show the dependencies version just calling a function that returns a string. You can see this function kind present in many libraries too.